sql/result.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sql
+
+// Result represents a query result. Depending on the statement type it represents a stream of rows or an update count.
+type Result interface {
```

Comment:
I have some remarks for the docs of this interface's implementation. Comments for this interface may have to be synced with that.

Reply:
Thanks https://github.com/hazelcast/hazelcast-go-client/pull/755/commits/7c41e65d201592ac535e1ae9744a6a2c2ad36820